### PR TITLE
Preparing for monthly release, 0.6.0.0

### PR DESCRIPTION
**Pull request by alanz**

### CODE REVIEW

**Comment by alanz** [comment]
cc @power-fungus and reviewers, please check the changelog for typos/other stupid errors on my part.

**Comment by power-fungus** [comment]
@alanz looks fine to me

File: Changelog.md [hunk 1]
```diff
@@ -1,3 +1,38 @@
...
+- documentation updates
+  - document reactorPidcache (@bubba)
+  - Add a note in README about dyld path for macOS builds (@kubum)
+  - document workaround for missing gmp library (@Rhywun)
+  - Change --recursive to --recurse-submodules when cloning
+    (@leifmetcalf)
+- Speed up CI on circleci (@bubba)
+- Build via make
+  - Recursively sync and update submodules in Makefile (@bubba)
+- buid via shake
```
[the remark this answers is not in the document]
Comment (alanz) [reply Changelog.md]: thanks. I guess I should proof read too.

**Comment by alanz** [comment]
@bubba can you do the release thing on github?

**Comment by bubba** [comment]
@alanz on it

**Pull request merged**